apps/website-25/src/components/courses/UnitFeedback.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useCallback, useState } from 'react';
+import useAxios from 'axios-hooks';
+import { isMobile } from 'react-device-detect';
+import { CTALinkOrButton, useAuthStore } from '@bluedot/ui';
+import { FaCircleCheck } from 'react-icons/fa6';
+import axios from 'axios';
+
+import { Unit } from '../../lib/api/db/tables';
+import { GetUnitFeedbackResponse, PutUnitFeedbackRequest } from '../../pages/api/courses/[courseSlug]/[unitId]/feedback';
+
+const Star: React.FC<{ filled: boolean }> = ({ filled }) => (
+  <svg viewBox="0 0 39 37" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path
+      d="M24.4794 11.4922C24.7341 12.008 25.1966 12.3851 25.744 12.5352L25.9833 12.585L37.2802 14.2363L29.1073 22.1963C28.665 22.6271 28.4482 23.2364 28.5145 23.8457L28.5321 23.9678L30.4608 35.2119L20.3593 29.8994C19.813 29.6123 19.167 29.5944 18.6083 29.8457L18.4979 29.8994L8.39636 35.2119L10.3251 23.9678C10.4364 23.3187 10.2216 22.6558 9.74988 22.1963L1.57703 14.2363L12.8739 12.585C13.5243 12.4898 14.0867 12.0816 14.3778 11.4922L19.4286 1.25879L24.4794 11.4922Z"
+      stroke="#FFAE36"
+      strokeWidth="2"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+      fill={filled ? '#FFC16A' : 'transparent'}
+    />
+  </svg>
+);
+
+type UnitFeedbackProps = {
+  unit: Pick<Unit, 'id' | 'courseSlug'>;
+};
+
+const FREE_RESPONSE_PLACEHOLDER = '• What would have made this unit more valuable to you?\n• What aspects did you find most challenging?\n• What aspects did you find particularly useful?';
+
+const UnitFeedback: React.FC<UnitFeedbackProps> = ({ unit }) => {
+  const auth = useAuthStore((s) => s.auth);
+
+  const [rating, setRating] = useState<number>(0);
+  const [hoverRating, setHoverRating] = useState<number>(0);
+  const [feedbackText, setFeedbackText] = useState<string>('');
+  const [error, setError] = useState<string | null>(null);
+
+  const { courseSlug, id: unitId } = unit;
+
+  const [{ data }, refetch] = useAxios<GetUnitFeedbackResponse>({
+    method: 'get',
+    url: `/api/courses/${courseSlug}/${unitId}/feedback`,
+    headers: {
+      Authorization: `Bearer ${auth?.token}`,
+    },
+  }, { manual: !auth });
+
+  const handleSubmit = useCallback(async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!rating) {
+      setError('Please select a star rating');
+      return;
+    }
+
+    setError(null);
+
+    try {
+      await axios.put<unknown, unknown, PutUnitFeedbackRequest>(
+        `/api/courses/${courseSlug}/${unitId}/feedback`,
+        {
+          overallRating: rating,
+          anythingElse: feedbackText,
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${auth?.token}`,
+          },
+        },
+      );
+
+      await refetch();
+    } catch (e) {
+      setError('Unexpected error');
```

Comment:
since this sets on the client, let's add more information to the error:
```suggestion
      setError('Unexpected error occurred. Please reach out to team@bluedot.org if this error persists.');
```